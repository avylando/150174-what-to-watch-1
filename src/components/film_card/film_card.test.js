import React from "react";
import renderer from "react-test-renderer";
import FilmCard from "./film_card.jsx";

it(`FilmCard correctly renders`, () => {
  const data = {
    title: `Bohemian Rhapsody`,
    imageSrc: `img/bohemian-rhapsody.jpg`,
  };

  const tree = renderer
    .create(
        <FilmCard data={data}/>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
