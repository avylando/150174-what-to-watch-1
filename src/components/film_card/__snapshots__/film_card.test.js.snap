// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmCard correctly renders 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
    style={
      Object {
        "zIndex": 1,
      }
    }
  >
    <img
      alt="Bohemian Rhapsody"
      height="175"
      src="img/bohemian-rhapsody.jpg"
      width="280"
    />
  </div>
  <div
    className="player"
    style={
      Object {
        "zIndex": -1,
      }
    }
  >
    <video
      className="player__video"
      poster="img/bohemian-rhapsody.jpg"
      src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
    />
    <button
      className="player__exit"
      style={
        Object {
          "display": "none",
        }
      }
      type="button"
    >
      Exit
    </button>
    <div
      className="player__controls"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <div
        className="player__controls-row"
      >
        <div
          className="player__time"
        >
          <progress
            className="player__progress"
            max="100"
            value={0}
          />
          <div
            className="player__toggler"
            style={
              Object {
                "left": "0%",
              }
            }
          >
            Toggler
          </div>
        </div>
        <div
          className="player__time-value"
        >
          0
        </div>
      </div>
      <div
        className="player__controls-row"
      >
        <button
          className="player__play"
          onClick={[Function]}
          type="button"
        >
          <svg
            height="19"
            viewBox="0 0 19 19"
            width="19"
          >
            <use
              xlinkHref="#play-s"
            />
          </svg>
          <span>
            Play
          </span>
        </button>
        <div
          className="player__name"
        >
          Bohemian Rhapsody
        </div>
        <button
          className="player__full-screen"
          type="button"
        >
          <svg
            height="27"
            viewBox="0 0 27 27"
            width="27"
          >
            <use
              xlinkHref="#full-screen"
            />
          </svg>
          <span>
            Full screen
          </span>
        </button>
      </div>
    </div>
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Bohemian Rhapsody
    </a>
  </h3>
</article>
`;
